x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/charts/common/settings.ts

```diff
@@ -50,3 +54,43 @@ export const seriesStyle = {
     visible: false,
   },
 };
+
+export function useChartSettings() {
```

Comment:
I'm trying to wrap my head around the usage of this 😄 so I have this question: Why does this need to be a hook if you're returning the value directly from `getChartSettings()`? I'm not sure but could this have unexpected side effects? Just some food for thought:

- You could rename `getChartSettings()` to `updateChartSettings()` and remove `return cs;`. This would mean changing all code that consumes the hook to react to `chartSettings` only and not rely on a value being returned immediately.
- If we leave the code as is it might be worth adding a comment why this combination of it being a hook and returning a value immediately is necessary. Maybe rename `getChartSettings()` to `updateAndGetChartSettings()`, with just `get` it might be unexpected that this function has side effects.
- And yet maybe another way to approach this: `getChartSettings()` could become a `useEffect()` itself so it would update on demand if `jobCreator` or `chartInterval` changes. Then you'd always get the most up to date `chartSettings` from the hook.

Reply:
I'm not surprised you questioned this. This ended up being the most fiddly and time consuming part of this refactor.
Originally had `getChartSettings` be a basic util function that took `jobCreator` and `chartInterval` as parameters. but i then decided that it could be a hook which could take advantage of the context, as per your last point. the problem here is that i then needed another `useEffect` in every file using this hook to trigger the `loadCharts` on a `chartSettings` change. and each of those files already have a lot of `useEffects`.

I think the solution is to go back to a normal, non-hook, util function that calculates and returns the `chartSettings`. Each file calling it will then need to have their own `useState` to wrap this so it can be passed into the `ChartGrid` component at the bottom of the file.
The fact that each file will need their own `useState` is the main reason i went with a custom hook, so they could share it. But I also need the calculated `chartSettings` immediately, hence why i was also returning a function that could supply it.

Reply:
Removed custom hook in 39d24efd61ccfa59d8c311dd8db3774d6efb2d1a